firedrake/slate/slac/compiler.py

```diff
@@ -41,18 +45,167 @@
+            it_type = cxt_kernel.original_integral_type
+            exp = cxt_kernel.tensor
+
+            if it_type not in supported_integral_types:
+                raise ValueError("Integral type '%s' not recognized" % it_type)
+
+            # Explicit checking of coordinates
+            coordinates = cxt_kernel.tensor.ufl_domain().coordinates
+            if coords is not None:
+                assert coordinates == coords, "Mismatching coordinates!"
+            else:
+                coords = coordinates
+
+            for split_kernel in cxt_kernel.tsfc_kernels:
+                indices = split_kernel.indices
+                kinfo = split_kernel.kinfo
+
+                # TSFC No-op kernels do not contain a kernel body. If this is
+                # the case, we don't create a function call.
+                if kinfo.kernel:
+                    # TODO: Implement subdomains for Slate tensors
+                    if kinfo.subdomain_id != "otherwise":
+                        raise NotImplementedError("Subdomains not implemented.")
+
+                    args = [c for i in kinfo.coefficient_map
+                            for c in self.coefficient(local_coefficients[i])]
+
+                    if kinfo.oriented:
+                        args.insert(0, cell_orientations_sym)
+
+                    if kinfo.integral_type in ["interior_facet",
+                                               "exterior_facet",
+                                               "interior_facet_vert",
+                                               "exterior_facet_vert"]:
+                        args.append(ast.FlatBlock("&%s" % it_sym))
+
+                    # Assembly calls within the macro kernel
+                    tensor = eigen_tensor(exp, self.temps[exp], indices)
+                    call = ast.FunCall(kinfo.kernel.name,
+                                       tensor,
+                                       coord_sym,
+                                       *args)
+                    assembly_calls[it_type].append(call)
+
+                    # Subkernels for local assembly (Eigen templated functions)
+                    kast = transformer.visit(kinfo.kernel._ast)
+                    templated_subkernels.append(kast)
+                    include_dirs.extend(kinfo.kernel._include_dirs)
+                    oriented = oriented or kinfo.oriented
+
+        self.assembly_calls = assembly_calls
+        self.templated_subkernels = templated_subkernels
+        self.include_dirs = list(set(include_dirs))
+        self.oriented = oriented
+
+    @property
+    def integral_type(self):
+        """Returns the integral type associated with a Slate kernel.
+
+        Note that Slate kernels (for now) are always of type 'cell' since
+        these are localized kernels for element-wise linear algebra. This
+        may change in the future if we want Slate to be used for LDG/CDG
+        finite element discretizations.
+        """
+        return "cell"
```

Comment:
Best not to add "we might do this in the future" stuff into the docstring, you could rather write:

```
integral_type = "cell"
"Integral type associated with a Slate kernel."